File: python/evals/autocontext_fixture/src/auth/session.py
"""SessionManager + create_session."""
import secrets
from src.config.settings import Settings


class SessionManager:
    def __init__(self):
        self.sessions = {}

    def get(self, token):
        return self.sessions.get(token)

    def invalidate(self, token):
        self.sessions.pop(token, None)


def create_session(email: str):
    token = secrets.token_urlsafe(32)
    settings = Settings()
    manager = SessionManager()
    manager.sessions[token] = {"email": email, "ttl": settings.session_ttl}
    return type("Session", (), {"token": token, "email": email})

File: python/evals/autocontext_fixture_js/src/auth/session.ts
// Session manager.
import { randomBytes } from "crypto";
import { loadConfig } from "../config/settings";

export class SessionManager {
  private sessions = new Map<string, any>();
  get(token: string) { return this.sessions.get(token); }
  invalidate(token: string) { this.sessions.delete(token); }
  put(token: string, data: any) { this.sessions.set(token, data); }
}

export function createSession(email: string) {
  const cfg = loadConfig();
  const token = randomBytes(24).toString("hex");
  return { token, email, ttl: cfg.sessionTtl };
}
